pkg/sources/reconciler/awssnssource/subscription.go

```diff
@@ -52,14 +52,14 @@ func (r *Reconciler) ensureSubscribed(ctx context.Context) error {
 	status := &src.(*v1alpha1.AWSSNSSource).Status
 
 	isDeployed := status.GetCondition(v1alpha1.ConditionDeployed).IsTrue()
-	url := status.Address.URL
 
 	// skip this cycle if the URL couldn't yet be determined
-	if !isDeployed || url == nil {
+	if !isDeployed || status.Address == nil || status.Address.URL == nil {
```

Comment:
+ for Anotine's comment - Isn't `IsDeployed` mean that Address is set?

Reply:
yes, removing that one.